masu/__init__.py

```diff
@@ -0,0 +1,70 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""App factory for Masu application."""
+import logging
+import os
+
+from flask import Flask
+from flask.logging import default_handler
+from flask_sqlalchemy import SQLAlchemy
+
+from masu.api.status import StatusView
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+logger.addHandler(default_handler)
+
+
+def create_app(test_config=None):
+    """
+    App factory for Flask application.
+
+    Args:
+        test_config (dict): A mapping of configuratons used for testing
+
+    Returns:
+        flask.app.Flask: The configured Flask application
+
+    """
+    app = Flask(__name__, instance_relative_config=True)
+
+    # Load configs
+    if test_config is None:
+        app.config.from_object('masu.config.Config')
+    else:
+        app.config.from_mapping(test_config)
+
+    # Logging
+    logger.setLevel(app.config.get('LOG_LEVEL', 'WARNING'))
+
+    try:
+        os.makedirs(app.instance_path)
+    # pylint: disable=invalid-name
+    except OSError as e:
+        logger.warning(e)
+
+    # Establish database
+    # pylint: disable=invalid-name, unused-variable
+
+    db = SQLAlchemy(app)  # noqa: F841
+
+    # Routes
+    app.add_url_rule('/status/', view_func=StatusView.as_view('show_status'))
```

Comment:
Don't know enough about Flask. Will the url path be the same as koku (i.e. `/api/v1/status/`)? I think it will make it easy if we are consistent from a health check/ops perspective.

Reply:
Good point. The quickest way, given the very few endpoints likely here is just to prefix the route with `/api/v1` directly in the url_rule here. I believe we can also achieve this using Flask Blueprints, but I don't think they play nice with the class based view I wrote. Very happy to change that, thought. I went that route to mirror the koku API endpoint.

Reply:
I'm fine with just prefixing with `/api/v1` for now.